allennlp/models/encoder_decoders/nlvr/nlvr_coverage_semantic_parser.py

```diff
@@ -107,6 +112,55 @@ def __init__(self,
             self._dynamic_cost_rate = dynamic_cost_weight["rate"]
         self._penalize_non_agenda_actions = penalize_non_agenda_actions
         self._last_epoch_in_forward: int = None
+        if initial_mml_model_file is not None:
+            archive = load_archive(initial_mml_model_file)
+            self._initialize_weights_from_archive(archive)
+
+    def _initialize_weights_from_archive(self, archive: Archive) -> None:
+        logger.info("Initializing weights from MML model.")
+        model_parameters = dict(self.named_parameters())
+        archived_parameters = archive.model.named_parameters()
+        for name, weights in archived_parameters:
+            if name in model_parameters:
+                if name == "_decoder_step._output_projection_layer.weight":
+                    # The dimensions differ for this parameter between the coverage model and
+                    # the direct model. In the direct model, this is of size
+                    # (decoder_output_dim + encoder_output_dim, action_embedding_dim),
+                    # whereas in the coverage model, it is
+                    # (decoder_output_dim + encoder_output_dim + checklist_size, action_embedding_dim)
+                    # We copy only the relevant part of the weights here.
+                    archived_projection_weights = weights.data
+                    new_weights = model_parameters[name].data.clone()
+                    new_weights[:, :-self.num_terminals] = archived_projection_weights
+                elif name == "_sentence_embedder.token_embedder_tokens.weight":
```

Comment:
Relying on this is pretty brittle, as you're assuming that there's a "tokens" embedder using the "tokens" namespace.  That's fine for now, but I'd at least note that in the docstring for the `initial_mml_model_file` parameter.  If there's a way you could crash if this assumption isn't met, too, that'd be good.

Reply:
I added a check to fail if the name is not in one of the two named parameters.